specs/merkle-tree.md

```diff
@@ -0,0 +1,158 @@
+
+### 2. Merkle tree leaves must be ordered in the tree by index and chain ID ascending
+Sphinx Merkle tree leaves use an explicit index value rather than an implicit value determined by the order in the tree. This is because the Sphinx Merkle tree is intended to be executed across multiple networks and not all leaves will be executed on all networks. As a result, it is possible to construct Sphinx Merkle trees that have leaves that follow various non-standard ordering schemes. These trees are executable on-chain as long as their leaves are submitted by order of their index. However, these trees may not be executable by all off-chain relayer implementations which may rely on an expected order.
+
+For example, the following Merkle tree would be valid as long as its leaves are submitted on-chain in ascending order by their index. This is true even though the tree was constructed with the order of the leaves reversed.
+```mermaid
+graph LR
+    A[Root] --> B[Hash]
+    A[Root] --> C[Hash]
+    B[Hash] --> D[Hash]
+    B[Hash] --> E[Hash]
+    C[Hash] --> F[Hash]
+    C[Hash] --> G[Hash]
+    D[Hash] --> I["Execute: Chain ID: 5, Index: 3, Data: 0xaa..."]
+    D[Hash] --> J["Execute: Chain ID: 5, Index: 2, Data: 0xbb..."]
+    E[Hash] --> K["Execute: Chain ID: 5, Index: 1, Data: 0xcc..."]
+    E[Hash] --> H["Approve: Chain ID: 5, Index: 0"]
+    F[Hash] --> M["Execute: Chain ID: 420, Index: 3, Data: 0xdd..."]
+    F[Hash] --> N["Execute: Chain ID: 420, Index: 2, Data: 0xee..."]
+    G[Hash] --> O["Execute: Chain ID: 420, Index: 1, Data: 0xff..."]
+    G[Hash] --> L["Approve: Chain ID: 420, Index: 0"]
+```
+
+Sphinx Merkle trees should follow a consistent ordering scheme to ensure they can be executed by a variety of executor implementations without issue. We impose the standard that the leaves should be ordered by index and chain ID ascending.
+
+So for example, if we intended to run some set of transactions on Goerli (5), OP Goerli (420), and Polygon Mumbai (80001). Then the leaves are expected to be ordered with the approval leaf for Goerli first, then all of the transaction leaves on Goerli in ascending order, then the approval leaf for OP Goerli, all the transaction leaves on OP Goerli in ascending order, and so on.
+
+### 3. Every chain ID and index combination must be unique within the tree
+Since the Sphinx Merkle tree leaves contain explicit indexes, it is also possible to construct trees that contain multiple leaves with the same index value and chain ID. However, only one of the leaves would be executable on-chain. Therefore, constructing a tree with multiple leaves with the same chain ID and index would make it ambiguous to the relayer which leaf should be executed, and would result in a tree that cannot be fully executed. So we impose the standard that in a given Sphinx Merkle tree there must be at most one leaf with each chain ID and index combination.
```

Comment:
I think this paragraph can be condensed into a single sentence. I think it's enough rationale to just say that this ensures there is an unambiguous ordering of leaves on each chain